pkg/validation/spec/responses.go

```diff
@@ -90,20 +90,28 @@ func (r ResponsesProps) MarshalJSON() ([]byte, error) {
 
 // UnmarshalJSON unmarshals responses from JSON
 func (r *ResponsesProps) UnmarshalJSON(data []byte) error {
-	var res map[string]Response
+	var res map[string]json.RawMessage
 	if err := json.Unmarshal(data, &res); err != nil {
 		return nil
```

Comment:
why do we return nil here?

Reply:
Also, I've tried to switch that to `return err`, but it still doesn't fail even when it should. Are there other occurences of errors not being propagated properly?

Reply:
> but it still doesn't fail even when it should. 

I'm not sure what you've tried, but it should fail if provided with a non-map element i.e. `["a", "b", "c"]`, or `123`.

you are right that this should not return nil